bindings/cpp/include/wtf/macros.h

```diff
@@ -117,6 +120,28 @@
           __WTF_INTERNAL_UNIQUE(__wtf_scope_eventn_)};                        \
   __WTF_INTERNAL_UNIQUE(__wtf_scopen_).Enter
 
+// Shortcut to trace a function in case you don't care really much about
+// the performance. Might add some insignificant overhead.
+// It will also replace colon ":" with hash sign "#" in function names
+// (see https://github.com/google/tracing-framework/issues/581)
+// Allowed Scopes: Within a function.
+// Usually you will place this at the very start of a function.
+//
+// Example:
+//   WTF_AUTO_FUNCTION();
+#define WTF_AUTO_FUNCTION()                                             \
+    static std::string                                                  \
+    __WTF_INTERNAL_UNIQUE(__wtf_func_name_) {__PRETTY_FUNCTION__};      \
+    do {                                                                \
+        static std::once_flag flag;                                     \
+        std::call_once(flag, []() {                                     \
```

Comment:
it looks like this should be protected as "single" threading doesn't support std::call_once (throwing exception)